Shared/Samples/Navigate route/NavigateRouteView.swift

```diff
@@ -0,0 +1,354 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import AVFoundation
+import SwiftUI
+
+struct NavigateRouteView: View {
+    /// The view model for this sample.
+    @StateObject private var model = Model()
+    
+    var body: some View {
+        MapView(
+            map: model.map,
+            viewpoint: model.viewpoint,
+            graphicsOverlays: model.graphicsOverlays
+        )
+        .onViewpointChanged(kind: .centerAndScale) { model.viewpoint = $0 }
+        .locationDisplay(model.locationDisplay)
+        .alert(
+            isPresented: Binding(get: { model.error != nil }, set: { _ in model.error = nil }),
+            presentingError: model.error
+        )
+        .overlay(alignment: .top) {
+            Text(model.statusText)
+                .padding(2)
+                .frame(maxWidth: .infinity, alignment: .leading)
+                .background(.ultraThinMaterial)
+        }
+        .task {
+            // Solves the route and sets the navigation.
+            await model.solveRoute()
+            model.setNavigation()
+        }
+        .toolbar {
+            ToolbarItemGroup(placement: .bottomBar) {
+                Button("Navigate") {
+                    model.isNavigatingRoute = true
+                }
+                .task(id: model.isNavigatingRoute) {
+                    guard model.isNavigatingRoute else { return }
+                    await model.startNavigation()
+                }
+                .disabled(model.isNavigateDisabled || model.isNavigatingRoute)
+                
+                Spacer()
+                
+                Button("Recenter") {
+                    model.autoPanMode = .navigation
+                }
+                .disabled(!model.isNavigatingRoute || model.autoPanMode == .navigation)
```

Comment:
Seems to me that `model.autoPanMode` only changes when the navigation is ongoing.
```swift
.disabled(model.autoPanMode == .navigation)
```
If you don't set the `model.autoPanMode` to `.off` in the initializer, but to `.navigation` either in the initializer or directly on the property's declaration, then I think this condition can be simplified.

It is OK if you don't think it can be removed.

Reply:
`Model.autoPanMode` can be initialized to `.navigation` but the button will still be enabled then when there's no location to snap to. I think the condition should remain.

![2023-03-28 14 45 16](https://user-images.githubusercontent.com/16397058/228373946-33c2b852-0edd-4da8-a3d0-858c5d1026e7.gif)